android/shopify_buy_plugin/src/main/java/com/shopify/unity/buy/UnityAndroidPayFragment.java

```diff
@@ -169,32 +177,72 @@ public void onMaskedWallet(final MaskedWallet maskedWallet) {
 
                 MailingAddressInput input = new MailingAddressInput(maskedWallet.getBuyerShippingAddress());
 
+                Logger.d("Masked wallet received");
+
                 if (sessionCallbacks != null) {
                     sessionCallbacks.onUpdateShippingAddress(input, new MessageCenter.MessageCallback() {
                         @Override
                         public void onResponse(String jsonResponse) {
-                            // TODO: Create a new pay cart with the updated shipping address and request full wallet
+                            UnityAndroidPayFragment.this.onUpdateShippingAddress(jsonResponse);
                         }
                     });
                 }
             }
 
             @Override
+            public void onFullWallet(FullWallet fullWallet) {
+                super.onFullWallet(fullWallet);
+                Logger.d("Full wallet received.");
+            }
+
+            @Override
             public void onWalletError(int requestCode, int errorCode) {
+                final String msg = "Wallet error: " + WalletErrorFormatter.errorStringFromCode(errorCode);
+                Logger.d(msg);
                 if (sessionCallbacks != null) {
                     sessionCallbacks.onError(WalletErrorFormatter.errorStringFromCode(errorCode));
                 }
             }
 
             @Override
             public void onWalletRequestCancel(int requestCode) {
+                super.onWalletRequestCancel(requestCode);
+                Logger.d("Wallet canceled");
                 if (sessionCallbacks != null) {
                     sessionCallbacks.onCancel();
                 }
             }
         });
     }
 
+    private void onUpdateShippingAddress(String jsonResponse) {
+        // TODO: Create a new pay cart with the updated shipping address and request full wallet
+        Logger.d("New cart data from Unity: " + jsonResponse);
+        try {
+            cart = newPayCart(AndroidPayCheckoutResponse.fromJsonString(jsonResponse));
```

Comment:
Nit: I would rename this to be more clear:

`payCartFromCheckoutResponse`

Reply:
👍

Reply:
https://github.com/Shopify/unity-buy-sdk/pull/321/commits/49eb13c9ec23d53f3b2eab71960d3bbd142a3ffb